CMakeLists.txt

```diff
@@ -78,14 +78,26 @@ add_subdirectory(src)
 #By including the file "BipedalLocomotion/Framework.h" it is possible to include all the headers automatically.
 include(AddBipedalLocomotionFrameworkTarget)
 
+set(FRAMEWORK_PUBLIC_DEPENDENCIES iDynTree)
+
+if (FRAMEWORK_HAS_YARP)
+  list(APPEND FRAMEWORK_PUBLIC_DEPENDENCIES YARP)
+endif()
+
+if (FRAMEWORK_HAS_Eigen3)
+  list(APPEND FRAMEWORK_PRIVATE_DEPENDENCIES Eigen3)
+endif()
```

Comment:
Actually this  would be a bit more robust if it depends on YARP only if the components that depends in a public way are enabled,  not only if YARP or Eigen3 is found in the system.

Reply:
Actually if you do not use  Eigen3 as an inclusion in the public headers, I don't think it make sense to have it as a private dependency as it is an header-only  dependency.

Reply:
> Actually this would be a bit more robust if it depends on YARP only if the components that depends in a public way are enabled

I thought about that at the beginning, but then I did not find any way to automatize this check. In principle, I would need to determine a set of dependencies depending on the compiled targets. Initially, I thought of adding an entry in the ``add_bipedal_library`` function (something like ``DEPENDENCIES YARP iDynTree``), but this would be a sort of duplicate of the targets to link and it is error-prone since it does not throw any error if you miss some of them.
The other option was to change the if condition to check directly if some of the enabled components have ``YARP`` enabled. But then, every time we add a new component we would need to change also those ``if``, with a high likelihood of forgetting one from time to time. Hence, I thought that the safest and easiest option was to propagate all the dependencies that were already found in the system.

> Actually if you do not use Eigen3 as an inclusion in the public headers, I don't think it make sense to have it as a private dependency as it is an header-only dependency.

Even if we compile statically?

Reply:
> The other option was to change the if condition to check directly if some of the components having YARP as dependency are enabled (like what you suggest). But then, every time we add a new component we would need to change also those if, with a high likelihood of forgetting one from time to time. Hence, I thought that the safest and easiest option was to propagate all the dependencies that were already found in the system.

That make sense, but can we use `FRAMEWORK_USE_YARP` instead of `FRAMEWORK_HAS_YARP`? So a user can explicitly disable it.

> > Actually if you do not use Eigen3 as an inclusion in the public headers, I don't think it make sense to have it as a private dependency as it is an header-only dependency.
> 
> Even if we compile statically?

An header only library is only used at compile time, and not at the linking time, so you safely link a static library that used internally Eigen3 completely ignoring the fact that Eigen3 has been used.

Reply:
> An header only library is only used at compile time, and not at the linking time, so you safely link a static library that used internally Eigen3 completely ignoring the fact that Eigen3 has been used.

It makes sense, I will remove it, thanks!

> That make sense, but can we use FRAMEWORK_USE_YARP instead of FRAMEWORK_HAS_YARP? So a user can explicitly disable it.

Deal! Should I use the same flag also here in the following lines? https://github.com/dic-iit/bipedal-locomotion-framework/blob/690cb30180554b5f8c83c40a18ffadd3573d9488/cmake/BipedalLocomotionFrameworkFindDependencies.cmake#L131-L141

Reply:
Using ``USE`` and removed ``Eigen`` as a private dependency (in the config file) in https://github.com/dic-iit/bipedal-locomotion-framework/pull/43/commits/30e2f065946ba35b62a0f93b5975b6b6d969a0ac